net-proxy/mitmproxy/mitmproxy-5.2.ebuild

```diff
@@ -68,6 +68,10 @@ python_prepare_all() {
 	# remove failing test
 	sed -i 's/test_get_version/_&/g' test/mitmproxy/test_version.py || die
 
+	# https://github.com/mitmproxy/mitmproxy/issues/4136
+	# https://bugs.gentoo.org/740336
+	rm test/mitmproxy/addons/test_termlog.py || die
+
```

Comment:
Even with this, the test phase fails for me. Does it work for you?
```
===================================== short test summary info ======================================
FAILED test/mitmproxy/net/test_tcp.py::TestSSLUpstreamCertVerificationWBadHostname::test_mode_none_should_pass_without_sni
FAILED test/mitmproxy/net/test_tcp.py::TestSSLUpstreamCertVerificationWValidCertChain::test_mode_strict_w_pemfile_should_pass
FAILED test/mitmproxy/net/test_tcp.py::TestSSLUpstreamCertVerificationWValidCertChain::test_mode_strict_w_confdir_should_pass
FAILED test/mitmproxy/proxy/test_server.py::TestHTTPSUpstreamServerVerificationWTrustedCert::test_verification_w_confdir
FAILED test/mitmproxy/proxy/test_server.py::TestHTTPSUpstreamServerVerificationWTrustedCert::test_verification_w_pemfile
================= 5 failed, 1475 passed, 2 xfailed, 2 warnings in 81.84s (0:01:21) =================
```
```
Exception in thread ProxyThread (*:34251):
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/test/mitmproxy/tservers.py", line 116, in run
    self.tmaster.run()
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/mitmproxy/master.py", line 108, in run
    self.run_loop(loop.run_forever)
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/mitmproxy/master.py", line 92, in run_loop
    loop = asyncio.get_event_loop()
  File "/usr/lib/python3.7/asyncio/events.py", line 644, in get_event_loop
    % threading.current_thread().name)
RuntimeError: There is no current event loop in thread 'ProxyThread (*:34251)'.

Exception in thread ProxyThread (*:40917):
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/test/mitmproxy/tservers.py", line 116, in run
    self.tmaster.run()
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/mitmproxy/master.py", line 108, in run
    self.run_loop(loop.run_forever)
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/mitmproxy/master.py", line 92, in run_loop
    loop = asyncio.get_event_loop()
  File "/usr/lib/python3.7/asyncio/events.py", line 644, in get_event_loop
    % threading.current_thread().name)
RuntimeError: There is no current event loop in thread 'ProxyThread (*:40917)'.

Exception in thread ProxyThread (*:43213):
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/test/mitmproxy/tservers.py", line 116, in run
    self.tmaster.run()
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/mitmproxy/master.py", line 108, in run
    self.run_loop(loop.run_forever)
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/mitmproxy/master.py", line 92, in run_loop
    loop = asyncio.get_event_loop()
  File "/usr/lib/python3.7/asyncio/events.py", line 644, in get_event_loop
    % threading.current_thread().name)
RuntimeError: There is no current event loop in thread 'ProxyThread (*:43213)'.

Exception in thread ProxyThread (*:37331):
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/test/mitmproxy/tservers.py", line 116, in run
    self.tmaster.run()
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/mitmproxy/master.py", line 108, in run
    self.run_loop(loop.run_forever)
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/mitmproxy/master.py", line 92, in run_loop
    loop = asyncio.get_event_loop()
  File "/usr/lib/python3.7/asyncio/events.py", line 644, in get_event_loop
    % threading.current_thread().name)
RuntimeError: There is no current event loop in thread 'ProxyThread (*:37331)'.

Exception in thread ProxyThread (*:41851):
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/test/mitmproxy/tservers.py", line 116, in run
    self.tmaster.run()
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/mitmproxy/master.py", line 108, in run
    self.run_loop(loop.run_forever)
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/mitmproxy/master.py", line 92, in run_loop
    loop = asyncio.get_event_loop()
  File "/usr/lib/python3.7/asyncio/events.py", line 644, in get_event_loop
    % threading.current_thread().name)
RuntimeError: There is no current event loop in thread 'ProxyThread (*:41851)'.

Exception in thread ProxyThread (*:34911):
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/test/mitmproxy/tservers.py", line 116, in run
    self.tmaster.run()
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/mitmproxy/master.py", line 108, in run
    self.run_loop(loop.run_forever)
  File "/var/tmp/portage/net-proxy/mitmproxy-5.2/work/mitmproxy-5.2/mitmproxy/master.py", line 92, in run_loop
    loop = asyncio.get_event_loop()
  File "/usr/lib/python3.7/asyncio/events.py", line 644, in get_event_loop
    % threading.current_thread().name)
RuntimeError: There is no current event loop in thread 'ProxyThread (*:34911)'.

 * ERROR: net-proxy/mitmproxy-5.2::testworld failed (test phase):
 *   Tests fail with python3.7
 * 
```

Reply:
Seems like these 5 tests pass on OpenSSL 1.1.1g, and fail with OpenSSL 1.1.1h. Application seems to work fine on 1.1.1h (other than the failing tests), and they were [disabled by the previous maintainer until I re-enabled them](https://github.com/gentoo/gentoo/blob/4b3c689acd8f1faf958f87ef60fc53f865e56705/net-proxy/mitmproxy/mitmproxy-5.1.1.ebuild#L76-L85). Will re-disable :x